tests/cpp/rhi/common/host_memory_pool_test.cpp

```diff
@@ -0,0 +1,29 @@
+#include "gtest/gtest.h"
+
+#include "taichi/rhi/common/host_memory_pool.h"
+
+namespace taichi::lang {
+
+#ifndef TARGET_OS_IPHONE
+// This code will only compile and run on non-iOS platforms
+
+TEST(HostMemoryPool, AllocateMemory) {
+  // fairly sure this leaks 1GB memory, but it's a test, so yeah
```

Comment:
Our CI machines are fairly memory constrained, can you shrink the pool size? If not possible this is probably fine.

Reply:
Yes, sure. It does mean removing `const` from the default allocator size though I think?

As far as being able to modify this from the test, despite being `private`, I realized I can make the test class a `friend`, and then I no longer need to make major changes to the allocator to make it testable. Making the test class a friend approximately aligns with gtest recommendations, https://google.github.io/googletest/advanced.html

![Screenshot 2025-05-06 at 6 45 08 AM](https://github.com/user-attachments/assets/ab02d622-e824-489d-8898-ed35e07a57fd)

Thoughts?